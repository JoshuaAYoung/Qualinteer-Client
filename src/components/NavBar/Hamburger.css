
/* change all of this with mobile only */
.navMenu{
  text-align: center;
  display: flex;
  flex-direction: column;
  height: 100vh;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  background-color: #FFEEA6;
  justify-content: space-evenly;
  padding: 13rem 0;
  z-index: 1;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
}

/* change with mobile only */
.navLink {
  color: black;
  text-decoration: none;
  font-family: 'Quicksand', sans-serif;
  font-weight: 700;
  font-size: 3.2rem;
  margin: 0 2rem;
}

.linkContainer {
  display: inline;
  position: static;
  margin: 0 auto;
  height: 6rem;
  border-bottom: .2rem solid #F4BD5B;
}

.linkBullet {
  margin: 0 .08rem;
  font-size: 1.5em;
  vertical-align: -.3rem
}

/* @media (max-width: 99.9rem) {
.mobileHamburger {
    position: fixed;
    top: 0;
    left: 0;
    overflow: hidden;
}
} */

.burgerButton {
  width: 2.5rem;
  height: 2.5rem;
  position: absolute;
  top: 1.9rem;
  right: 2rem;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -o-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: .5s ease-in-out;
  -moz-transition: .5s ease-in-out;
  -o-transition: .5s ease-in-out;
  transition: .5s ease-in-out;
  cursor: pointer;
  z-index: 1;
}

.burgerButton span {
  display: block;
  position: absolute;
  height: .2rem;
  width: 100%;
  background: black;
  border-radius: .9rem;
  opacity: 1;
  left: 0;
  -webkit-transform: rotate(0deg);
  -moz-transform: rotate(0deg);
  -o-transform: rotate(0deg);
  transform: rotate(0deg);
  -webkit-transition: .25s ease-in-out;
  -moz-transition: .25s ease-in-out;
  -o-transition: .25s ease-in-out;
  transition: .25s ease-in-out;
}

.burgerButton span:nth-child(1) {
  top: 0;
  -webkit-transform-origin: left center;
  -moz-transform-origin: left center;
  -o-transform-origin: left center;
  transform-origin: left center;
}

.burgerButton span:nth-child(2) {
  top: .9rem;
  -webkit-transform-origin: left center;
  -moz-transform-origin: left center;
  -o-transform-origin: left center;
  transform-origin: left center;
}

.burgerButton span:nth-child(3) {
  top: 1.8rem;
  -webkit-transform-origin: left center;
  -moz-transform-origin: left center;
  -o-transform-origin: left center;
  transform-origin: left center;
}

.trueBurger span:nth-child(1) {
  -webkit-transform: rotate(45deg);
  -moz-transform: rotate(45deg);
  -o-transform: rotate(45deg);
  transform: rotate(45deg);
  left: .4rem;
}

.trueBurger span:nth-child(2) {
  width: 0%;
  opacity: 0;
}

.trueBurger span:nth-child(3) {
  -webkit-transform: rotate(-45deg);
  -moz-transform: rotate(-45deg);
  -o-transform: rotate(-45deg);
  transform: rotate(-45deg);
  left: .4rem;
}

.fadeMenuHidden {
  display: none;
}

@-webkit-keyframes fadeMenuDown {
  from {
    opacity: 0;
    -webkit-transform: translate3d(0, -100%, 0);
    transform: translate3d(0, -100%, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}

@keyframes fadeMenuDown {
  from {
    opacity: 0;
    -webkit-transform: translate3d(0, -100%, 0);
    transform: translate3d(0, -100%, 0);
  }

  to {
    opacity: 1;
    -webkit-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}

.fadeMenuDown {
  -webkit-animation-name: fadeMenuDown;
  animation-name: fadeMenuDown;
}


@-webkit-keyframes fadeMenuUp {
  from {
    opacity: 1;
  }

  to {
    opacity: 0;
    -webkit-transform: translate3d(0, -100%, 0);
    transform: translate3d(0, -100%, 0);
  }
}

@keyframes fadeMenuUp {
  from {
    opacity: 1;
  }

  to {
    opacity: 0;
    -webkit-transform: translate3d(0, -100%, 0);
    transform: translate3d(0, -100%, 0);
  }
}

.fadeMenuUp {
  -webkit-animation-name: fadeMenuUp;
  animation-name: fadeMenuUp;
}

@-webkit-keyframes slideMenuDown {
  from {
    -webkit-transform: translate3d(0, -100%, 0);
    transform: translate3d(0, -100%, 0);
    visibility: visible;
  }

  to {
    -webkit-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}

@keyframes slideMenuDown {
  from {
    -webkit-transform: translate3d(0, -100%, 0);
    transform: translate3d(0, -100%, 0);
    visibility: visible;
  }

  to {
    -webkit-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }
}

.slideMenuDown {
  -webkit-animation-name: slideMenuDown;
  animation-name: slideMenuDown;
}





@-webkit-keyframes slideMenuUp {
  from {
    -webkit-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }

  to {
    visibility: hidden;
    -webkit-transform: translate3d(0, -100%, 0);
    transform: translate3d(0, -100%, 0);
  }
}

@keyframes slideMenuUp {
  from {
    -webkit-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }

  to {
    visibility: hidden;
    -webkit-transform: translate3d(0, -100%, 0);
    transform: translate3d(0, -100%, 0);
  }
}

.slideMenuUp {
  -webkit-animation-name: slideMenuUp;
  animation-name: slideMenuUp;
}
