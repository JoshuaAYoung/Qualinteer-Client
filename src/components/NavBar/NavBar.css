.navBar {
  background: #FFEEA6;
  width: 100%;
  position: fixed;
  height: 6rem;
  z-index: 999;
}

.logoLink {
  height: 100%;
  display: inline-block;
}

.mainLogo {
  width: 3.2rem;
  display: inline-block;
  margin-left: 2rem;
  vertical-align: -.8rem;
}

.navName {
  font-family: 'Quicksand', sans-serif;
  font-weight: 700;
  font-size: 2.3rem;
  line-height: 3.2rem;
  display: inline-block;
  letter-spacing: -.1rem;
  margin-left: .7rem;
  font-style: none;
  color: black;
  padding-right: 1rem;
}

