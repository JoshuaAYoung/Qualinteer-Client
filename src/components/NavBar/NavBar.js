import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import VableContext from '../../context/VableContext';
import './NavBar.css';
import TokenService from '../../services/token-service'
import './Hamburger.css'

export default class NavBar extends Component {
  static contextType = VableContext;

  state = {
    showBurger: false,
    animationClass: 'Hidden'
  }

  hideHamburger = () => {
    this.setState({
      showBurger: false,
      animationClass: 'Hidden'
    });
  }

  toggleHamburger = () => {
    if (this.state.showBurger === false || this.state.animationClass === 'Hidden') {
      this.setState({
        showBurger: true,
        animationClass: 'Down'
      })
    }
    else {
      this.setState({
        showBurger: false,
        animationClass: 'Up'
      })
    }
  }

  handleLogout = () => {
    this.hideHamburger();
    TokenService.clearAuthToken();
    this.context.setUserType('');
    this.context.setUser({});
    this.props.history.push('/');
  }

  linkSeparator = (<span className='linkBullet'>&bull;</span>)

  userLinks = () => {
    if (this.context.userType === 'volunteer' && TokenService.hasAuthToken()) {
      return (
        <>
          <div className='linkContainer'>
            <NavLink
              to='/'
              onClick={() => this.handleLogout()}
              className='navLink'>
              Log Out.
          </NavLink>
          </div>
        </>
      )
    }
    else if (this.context.userType === 'organization' && TokenService.hasAuthToken()) {
      return (
        <>
          <div className='linkContainer'>
            <NavLink
              to='/opportunitypost'
              className='navLink'
              onClick={() => this.hideHamburger()}>
              Recruit Help.
          </NavLink>
          </div>
          <div className='linkContainer'>
            <NavLink
              to='/'
              className='navLink'
              onClick={() => this.hideHamburger()}>
              Log Out.
          </NavLink>
          </div>
        </>
      )
    }
    else {
      return (
        <>
          <div className='linkContainer'>
            <NavLink
              to='/login'
              className='navLink'
              onClick={() => this.hideHamburger()}>
              Log In.
          </NavLink>
          </div>
          {/* {this.linkSeparator} */}
          <div className='linkContainer'>
            <NavLink
              to='/register'
              className='navLink'
              onClick={() => this.hideHamburger()}>
              Sign Up.
          </NavLink>
          </div>
        </>
      )
    }
  }

  render() {
    return (
      <div className='navContainer'>
        <div className='navBar'>
          <NavLink
            to='/'
            className='logoLink'
            onClick={() => this.hideHamburger()}>
            <img src='/assets/v-able-logo.svg' alt='v-able logo' className='mainLogo' />
            <h1 className='navName'>v<span className='inlineBullet'>&bull;</span>able</h1>
          </NavLink>
          <div className='hamburgerContainer'>
            <div
              className={`burgerButton ${this.state.showBurger}Burger`}
              onClick={() => this.toggleHamburger()}>
              <span></span>
              <span></span>
              <span></span>
            </div>
          </div>
        </div>
        <div className={`navMenu fadeMenu${this.state.animationClass}`}>
          {this.userLinks()}
          {/* {this.linkSeparator} */}
          <div className='linkContainer'>
            <NavLink
              to='/opportunityboard'
              className='navLink'
              onClick={() => this.hideHamburger()}>
              Opportunity Board.
          </NavLink>
          </div>
        </div>
      </div >
    );
  }
}
