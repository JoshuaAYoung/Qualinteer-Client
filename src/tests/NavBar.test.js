import React from 'react';
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NavBar from '../components/NavBar/NavBar'
import { BrowserRouter } from 'react-router-dom'

describe('NavBar Component', () => {

  it('NavBar Smoke Test', () => {
    const div = document.createElement('div')

    ReactDOM.render(<BrowserRouter><NavBar /></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('NavBar Snapshot Test', () => {
    const wrapper = shallow(
      <NavBar />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})