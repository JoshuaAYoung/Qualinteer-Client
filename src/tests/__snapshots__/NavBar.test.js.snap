// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component NavBar Snapshot Test 1`] = `
<div
  className="menuContainer"
>
  <div
    className="navMenu fadeMenuHidden"
  >
    <div
      className="linkContainer"
    >
      <NavLink
        className="navLink"
        onClick={[Function]}
        to="/"
      >
        Home
      </NavLink>
    </div>
    <span
      className="linkBullet"
    >
      •
    </span>
    <div
      className="linkContainer"
    >
      <NavLink
        className="navLink"
        onClick={[Function]}
        to="/login"
      >
        Log In
      </NavLink>
    </div>
    <span
      className="linkBullet"
    >
      •
    </span>
    <div
      className="linkContainer"
    >
      <NavLink
        className="navLink"
        onClick={[Function]}
        to="/register"
      >
        Sign Up
      </NavLink>
    </div>
    <span
      className="linkBullet"
    >
      •
    </span>
    <div
      className="linkContainer"
    >
      <NavLink
        className="navLink"
        onClick={[Function]}
        to="/opportunityboard"
      >
        Opportunity Board
      </NavLink>
    </div>
  </div>
  <div
    className="navBarContainer"
  >
    <NavLink
      className="logoLink"
      onClick={[Function]}
      to="/"
    >
      <img
        alt="v-able logo"
        className="mainLogo"
        src="/assets/v-able-logo.svg"
      />
      <h1
        className="navName"
      >
        v
        <span
          className="inlineBullet"
        >
          •
        </span>
        able
      </h1>
    </NavLink>
    <div
      className="hamburgerContainer"
    >
      <div
        className="burgerButton falseBurger"
        onClick={[Function]}
      >
        <span />
        <span />
        <span />
      </div>
    </div>
  </div>
</div>
`;
